server/app/controllers/availabilities_controller.rb

```diff
@@ -0,0 +1,137 @@
+class AvailabilitiesController < ApplicationController
+  before_action :confirm_user_logged_in
+  before_action :confirm_availability_exists, only: [:show, :update, :delete]
+
+  # GET /availabilities
+  def index
+    if current_user.usertype == "admin"
+      render json: {
+               availabilities: Availability.all,
+             }, status: :ok
+    else
+      render json: {
+               availabilities: current_user.availabilities,
+             }, status: :ok
+    end
+  end
+
+  # GET /availabilities/1
+  def show
+    @availability = Availability.find(params[:id])
+    if !confirm_requester_is_owner_or_admin(@availability.user.id)
+      return
+    end
+    render json: {
+             availability: @availability,
+           }, status: :ok
+  end
+
+  # POST /availabilities/
+  def create
+    if !confirm_requester_is_admin_or_rep_or_volunteer
+      return
+    end
+    params = availability_params.to_h
+    # If owner key is not provided, use the owner key provided by the requester
+    if !params.key?("user_id")
```

Comment:
Just to be sure - does this allow any admin/rep/volunteer to create availability for any other user? (I don't think that's the intended functionality)

Reply:
+ Same question for the other routes

Reply:
Good catch should fix this I'll render an error if it's not an admin

Reply:
For other routes, you should be able to get, update, delete availability only if you are admin or an owner of that availability